plugins/outputs/signalfx/signalfx.go

```diff
@@ -281,25 +294,79 @@ func (s *SignalFx) shouldSkipMetric(metricName string, metricTypeString string,
 	return false
 }
 
-func (s *SignalFx) emitDatapoints(datapoints []*datapoint.Datapoint) {
-	err := s.client.AddDatapoints(s.ctx, datapoints)
-	if err != nil {
-		log.Println("E! Output [signalfx] ", err)
+func (s *SignalFx) emitDatapoints() {
+	var buf []*datapoint.Datapoint
+	for {
+		select {
+		case dp := <-s.dps:
+			buf = append(buf, dp)
+			s.fillAndSendDatapoints(buf)
+			buf = buf[:0]
+		}
 	}
 }
 
-func (s *SignalFx) emitEvents(events []*event.Event) {
-	err := s.client.AddEvents(s.ctx, events)
-	if err != nil {
-		log.Println("E! Output [signalfx] ", err)
+func (s *SignalFx) fillAndSendDatapoints(buf []*datapoint.Datapoint) {
+	for {
+		select {
```

Comment:
need to watch the done channel as well to close the datapoints routine. Similar to how you did in the events routine.